traffic_portal/app/src/common/modules/notifications/notifications.tpl.html

```diff
@@ -18,7 +18,8 @@
 -->
 
 <div id="notificationsContainer">
-    <div class="alert alert-info" ng-repeat="n in notifications">
+    <div ng-show="!dismissedNotifications.includes(n.id)" class="alert alert-dismissable alert-info" ng-repeat="n in notifications | orderBy: 'cdn'">
```

Comment:
This should be using `ng-if`, since a notification can't be un-dismissed

Reply:
maybe i'll try ng-if but remember that notificaitons get refreshed so sometimes it needs to dynamically re-dismiss if you dismissed it earlier...

Reply:
Do they? Then it's probably a wash in performance gains between the DOM insertion/deletion work and the creation/deletion of data binding event handlers and DOM slots.

I'd guess - without measuring - it'd save memory to use `ng-if` but save some processing time to use `ng-show`. Probably not a big deal either way, but generally I tend to prefer `ng-if` because any given DOM insertion work is typically far less disruptive on a large scale than the memory usage of child scopes.

Reply:
right. i prefer ng-if as well. i'll use it if it works. tbh i only tried ng-show as i was thinking it was needed.

Reply:
switched to ng-if